src/utils/createCompactableHOC.js

```diff
@@ -1,25 +1,29 @@
 /* eslint-disable no-param-reassign */
 
 import { isMapperComponent as isCompacted } from './createHOCFromMapper'
-import createSymbol from './createSymbol'
+import WeakMap from './WeakMap'
 
-const compactable = createSymbol('compactable')
+const allCompactableComponents = new WeakMap()
 const isCompactable = Component =>
-  typeof Component === 'function' && Component[compactable]
+  typeof Component === 'function' && allCompactableComponents.has(Component)
```

Comment:
Don't know if it makes sense to get rid of `typeof Component === 'function'` (wasn't sure what it was guarding against/optimizing previously)?

Reply:
IMO, it was defensive programming (so that `isCompactable(null)` would return `false` instead of throwing a `TypeError` for instance).

I think we can probably get rid of it now that we're using a `WeakMap`.

Reply:
@matthieuprat ok removed `typeof` check here and in `isMapperComponent()`